### ISSUE TITLE
Change the specification of Clean Up

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#2039]
Prerequisites: #1925

### Summary

Change the cleanup spec. they will all be automated. However, an option will be added for users who prefer the old spec.

<details>
<summary>Expected changes (Web page)</summary>

#### (1/3) Add "Use Clean Up button" to Suppressed legacy features

![image](https://user-images.githubusercontent.com/27724847/219259615-5af7fbbf-04cf-4608-afeb-e28ba552fee7.png)

#### (2/3) Remove "Unnecessary data" from Internal Details

![image](https://user-images.githubusercontent.com/27724847/219259287-fffafca4-a911-486e-a99b-1602d2703504.png)

#### (3/3) Remove "Clean Up"

**Appears as before when "Use Clean Up button" is enabled**.

![image](https://user-images.githubusercontent.com/27724847/219266440-22dca89a-c1a1-45f1-add0-381308771e8f.png)

</details>

Specifically, all previous cleanup operations will be changed to run as part of the Scan process.

### Consideration

The previous spec wasn't completely evil.

 - The handling policy for most of Subsonic's storage is "append". Database deletion is logical deletion, and Lucene data deletion is not implemented.
   - Therefore, if the user does not manage anything, the storage may become bloated as the music data is deleted or changed. 
   - In Airsonic, Lucene data deletion has been implemented, and missearches have been somewhat improved. Jpsonic takes it even further.
 - Subsonic's specifications are not to be criticized at all.
   - There are times when such a design is intentionally made. If you want to avoid doing a lot of hard deletes in the main flow, or if you want to avoid hard disk fragmentation. Indeed, Lucene's data is designed to grow infinitely. However, it can be refreshed by stopping the server, deleting the Lucene directory, and scanning again.
   - The server's capability characteristics at the time this specification was implemented should be considered.

However, we can also think of it this way.

 - It is very rare for a media server to have a function called physical deletion in addition to scanning.
 - We want maintenance to be as easy as possible. We should be able to use it in the best condition even if you don't understand the data specification.
 - A well-thought-out library can assume that very few files will be changed or moved. All the user has to do is add dozens of songs to the existing library of tens of thousands of songs on a regular basis. Based on the most common use cases, uncommon cleanup buttons are distracting and confusing for newbies.
 - Extra queries will be issued on new registrations and updates to maintain the deferred hard delete specification. This is because it is necessary to check the existence of data before the deleted data is completely physically deleted. This cost may not be large, but of course it depends on the amount of data. In short, in theory some operations slow down as the amount of data increases. In normal use cases of fragmentation do not occur often, and the implementation to avoid it becomes a performance barrier.
   - There are some implementations that may affect performance, such as when creating random lists of songs.

Based on common use cases, it makes some sense to automate cleanup. On the other hand, there is no need to completely change or remove the previous specification, and there are reasons to leave it as an option.

**Comment by tesshucom** [comment]
This fix was scheduled for v111.7.0 because the amount of tasks in v111.6.0 is so bloated. But still, it seems better to push this fix into v111.6.0.

 - Personally (as a user) I dislike the cleanup specification as it is very annoying.
 - Some features may not work properly without cleanup. In theory it should follow the soft delete flag, but in reality not all implementations adhere to it.
   - I don't think you see many SQL deficiencies in legacy implementations. A relatively weak point is the link between Lucene and the DB. Until you clean up, Shuffle or Genre etc may behave badly for logical deletion of data.
   - For example #351.
   - v111.6.0 includes some fixes to handle genres more accurately, but they often don't apply without cleanup after scanning
